app/services/publish_reporting_service.rb

```diff
@@ -0,0 +1,123 @@
+class PublishReportingService
+  def initialize(recruitment_cycle_scope: RecruitmentCycle)
+    @courses = recruitment_cycle_scope.courses
+    @providers = recruitment_cycle_scope.providers
+  end
+
+  class << self
+    def call(recruitment_cycle_scope:)
+      new(recruitment_cycle_scope: recruitment_cycle_scope).call
+    end
+  end
+
+  def call
+    {
+      users: user_breakdown,
+      providers: provider_breakdown,
+      courses: course_breakdown,
+    }
+  end
+
+  private_class_method :new
```

Comment:
Why be strict about this? It's not like `self.call` does anything magical.

Reply:
so why not strict? why do you want to hold on to an instance?